easybuild/tools/github.py

```diff
@@ -718,12 +761,181 @@ def update_pr(pr, paths, commit_msg=None):
     print_msg(msg, log=_log, prefix=False)
 
 
+def check_github():
+    """
+    Check status of GitHub integration, and report back.
+    * check whether GitHub username is available
+    * check whether a GitHub token is available, and whether it works
+    * check whether git and GitPython are available
+    * check whether push access to own GitHub repositories works
+    * check whether creating gists works
+    * check whether location to local working directories for Git repositories is available (not strictly needed)
+    """
+    # start by assuming that everything works, individual checks will disable action that won't work
```

Comment:
makes the implementation less clean imho, now I can just set one (or more) of the `status` entries to `False` if a check fails...

for some of these, multiple checks must pass